shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -668,32 +672,64 @@ public long configureForTextureLayerComposition(
     return textureId;
   }
 
+  /**
+   * Translates an original touch event to have the same locations as the ones that Flutter
+   * calculates (because original + flutter's - original = flutter's).
+   *
+   * @param originalEvent The saved original input event.
+   * @param pointerCoords The coordinates that Flutter thinks the touch is happening at.
+   */
+  private void translateNonVirtualDisplayMotionEvent(
+      MotionEvent originalEvent, PointerCoords[] pointerCoords) {
+    if (pointerCoords.length < 1) {
+      return;
+    }
+
+    float xOffset = pointerCoords[0].x - originalEvent.getX();
+    float yOffset = pointerCoords[0].y - originalEvent.getY();
+
+    originalEvent.offsetLocation(xOffset, yOffset);
+  }
+
   @VisibleForTesting
   public MotionEvent toMotionEvent(
       float density, PlatformViewsChannel.PlatformViewTouch touch, boolean usingVirtualDiplay) {
     MotionEventTracker.MotionEventId motionEventId =
         MotionEventTracker.MotionEventId.from(touch.motionEventId);
     MotionEvent trackedEvent = motionEventTracker.pop(motionEventId);
 
+    // Pointer coordinates in the tracked events are global to FlutterView
```

Comment:
I have a deeper question about this change:

If the original MotionEvent had view-local coordinates why can't we just deliver the original MotionEvent here? Let me explain below:

When someone touches a Platform View the following happens:

1) MotionEvent (O) is delivered directly to the PlatformView. We stash it on the side in our "motionEventTracker".
2) Transform MotionEvent into a Flutter TouchEvent
3) TouchEvent is delivered to Flutter Widgets and eventually to the PlatformView widget.
4) We end up back in this code to (finally) deliver the event to the Android View.

My intuition would be that in step (1) the events in O have the correct coordinates. And then in (4) we deliver O to the view.

In what cases do we see a non-zero offset? Help me understand intuitively what's happening here.

Reply:
There is something weird going on here for sure. The reporting issue has a repro, which includes an example of a temporary fix that forces the platform view to refresh. In investigating why this temporarily fixes the issue, I printed out the original motion events and our calculated locations (and some more locations from framework code). 

It turns out refreshing the webview changes the location of the _**original motion event**_ that we store. Still need to investigate further as to why this would be the case, but it wasn't what I was originally expecting at all.

Reply:
It looks like probably the reason why we can't deliver the motion events as is in 4 is that the Platform view in question believes it is at `(0,0)` (thats the output of `platformView.getView.getX()` and `getY()`). Still not sure why that is though!

Reply:
Ok, I think I actually know the answer here now. `MotionEvent`s in general are relative to the view they are delivered to. So this particular motion event is global to the `FlutterView`, but still relative to a view. 

When we deliver the motion event in (4), we are then delivering it to an inner view, and as such need to make it relative to the inner view, which is why we need to offset. Does that make sense?

Reply:
(Talked with @johnmccutchan offline and we still aren't sure this reasoning is 100% correct, but landing regardless as the fix works better than what we have)